Move Corporate information pages from government-frontend

deborahchua opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

We want to handle the rendering of `corporate_information_page` documents from government-frontend to frontend.

## Why

[Trello card](https://trello.com/c/EKDXTDsm/398-move-document-type-corporateinformationpage-from-government-frontend-to-frontend), [Jira issue PNP-6452](https://gov-uk.atlassian.net/browse/PNP-6452)

## How

By moving the routes and necessary code to get the pages rendering in Frontend.

## Page examples
- /government/organisations/forensic-science-regulator/about (contents list with additional corporate information heading)
- /government/organisations/forensic-science-regulator/about/accessible-documents-policy (contents list without additional corporate information heading)
- /government/organisations/forensic-science-regulator/about/personal-information-charter (no contents list)
- /government/organisations/companies-house/about.cy (has Welsh translation)
- /government/organisations/department-for-environment-food-rural-affairs/about/modern-slavery-statement (no contents list)

## Screenshots

|government-frontend (before)| frontend (after)|
|-------|--------|
|![Screenshot 2025-01-22 at 11 38 56](https://github.com/user-attachments/assets/1f9c67dd-0767-4dc7-9e70-ed7b646924a8)|![Screenshot 2025-01-22 at 11 39 11](https://github.com/user-attachments/assets/b016d743-1d0c-49d7-92b5-b59bbd39335b)|
|![Screenshot 2025-01-22 at 11 43 36](https://github.com/user-attachments/assets/25a32919-0bdc-4abd-aa67-1a4ba960b768)|![Screenshot 2025-01-22 at 11 43 44](https://github.com/user-attachments/assets/e698a3b8-8780-4695-9818-a6d3ea81ce18)|
|![Screenshot 2025-01-22 at 11 45 42](https://github.com/user-attachments/assets/31bbae3c-a478-490d-9c03-f51eb77bc140)|![Screenshot 2025-01-22 at 11 45 52](https://github.com/user-attachments/assets/d7bed40e-f8c7-4ad3-95a6-bce331eaf67c)|
|![Screenshot 2025-01-22 at 14 47 07](https://github.com/user-attachments/assets/aaf150fa-46eb-49e6-bc66-90086e154a4e)|![Screenshot 2025-01-22 at 14 46 32](https://github.com/user-attachments/assets/aea3d6e7-6711-4af4-99fc-66e9c9293de1)|
|![Screenshot 2025-02-06 at 13 31 14](https://github.com/user-attachments/assets/4e7bcaf4-4fe3-466b-b410-2f97a423759d)|![Screenshot 2025-02-06 at 13 31 05](https://github.com/user-attachments/assets/b41528ea-ea9d-4abd-ad77-a9619cf521c0)|

deborahchua committed:
Update README.md and dev controller examples

deborahchua committed:
Add corporate_information_page routes, initial controller and view

The view has been directly copied over from government-frontend.
It will be updated to match the new data structure in a later
commit.

All occurences of `@content_item` have been updated to `content_item`.

Audit trail:
- https://github.com/alphagov/government-frontend/blob/8d2872c29486c4806cdd134cf1eb0a931f2a146c/app/views/content_items/corporate_information_page.html.erb

deborahchua committed:
Call Metadata component directly in the view

This keeps it consistent with the approach in other document types,
which call the metadata component directly. It does not include the
description tag which needs to be added manually for now.

It has also been updated to use `extra_headers`, which ensures that
the open graph meta tags are added to the page source correctly.

deborahchua committed:
Add locale keys for corporate information page content

These are displayed in the "Corporate information" section at the
bottom of the page if those pages are available.

Ran rake "consolidation:copy_translation[corporate_information_page, formats.corporate_information_page]"

Audit trail:
https://github.com/alphagov/government-frontend/tree/e7b9db33536f487991ba907502080e0625a066cc/config/locales

deborahchua committed:
Add basic model, controller and presenter

deborahchua committed:
Add default_organisation method to CorporateInformationPage model

This method gets the organisation based on the organisation ID from
the details hash. We can easily access the logo/brand information
now that we have modelled the Organisation and Logo, so we can pass
this directly to the `organisation_logo` publishing component in the view.

- We no longer need the hack to overide the executive order crest,
as this is already dealt with in Whitehall
- We not longer need to specify the brand class, as the `organisation_logo`
component takes care of this

Audit trail:
- https://github.com/alphagov/government-frontend/blob/4c38f1d3d8600373fe49ab506c758ee05e4909a3/app/presenters/corporate_information_page_presenter.rb#L33-L42
- https://github.com/alphagov/frontend/blob/1fe9d3e15cabe1f373314088449e63a2d35a02a3/app/models/organisation.rb
- https://github.com/alphagov/government-frontend/blob/4c38f1d3d8600373fe49ab506c758ee05e4909a3/app/presenters/content_item/organisation_branding.rb
- https://github.com/alphagov/government-frontend/blob/4c38f1d3d8600373fe49ab506c758ee05e4909a3/test/presenters/content_item/organisation_branding_test.rb

deborahchua committed:
Add system and request tests

Audit trail:

- https://github.com/alphagov/government-frontend/blob/8d2872c29486c4806cdd134cf1eb0a931f2a146c/test/integration/corporate_information_page_test.rb

KludgeKML commented on spec/components/contents_list_with_body_spec.rb:
```diff
@@ -0,0 +1,56 @@
+
+  it "renders nothing without a block" do
+    render_component(contents: contents_list)
+    expect(rendered).to be_empty
```
Could you add a space before the first expectation here? (a sort of non-lintable convention we have)

KludgeKML commented on spec/components/contents_list_with_body_spec.rb:
```diff
@@ -0,0 +1,56 @@
+
+  it "yields the block without contents data" do
+    render_component({}) { block }
+    expect(rendered).to include(block)
```
Same here

KludgeKML commented on spec/components/contents_list_with_body_spec.rb:
```diff
@@ -0,0 +1,56 @@
+  it "does not apply the sticky-element-container data-module without contents data" do
+    render_component(contents: contents_list) { block }
+
+    expect(rendered).to have_css("#contents[data-module='sticky-element-container']")
```
Is this test right? Test text says "does not apply the data module", but the expectation seems to thing it would still be there.

deborahchua replied:
Ah good spot thanks both! I copied the test over from government-frontend incorrectly - https://github.com/alphagov/government-frontend/blob/842a887dc0c83311a9ead2daa08bd430ad0ac347/test/components/contents_list_with_body_test.rb#L34-L37

KludgeKML commented on app/models/concerns/contents_list.rb:
```diff
@@ -0,0 +1,104 @@
+    end
+
+    def contents_items
+      extract_headings_with_ids
```
This line could do with a test - unless it's never expected to be called (always overwritten by the class the module is included in) - in which case we could maybe remove it entirely?

deborahchua replied:
I had a look at the test coverage and [the method itself](https://github.com/alphagov/frontend/pull/4608/commits/ef5ef221c521c6b84d0bf8ec4e809aab5a898ea3#diff-73e131976ad489a72d889fad2320885cced2b1c302b0c89373ea30d874f3cd4eR44-R50) is covered but not this specific line as `contents_items` is always overwritten for this document.

Some of the other document types don't overwrite this method, so this line should hopefully be covered by their tests when they are moved across.

KludgeKML commented on app/models/concerns/contents_list.rb:
```diff
@@ -0,0 +1,104 @@
+        return true if element.name == "div" && element["class"] == "img"
+
+        element = element.next_element
+        return false if element.nil?
```
This line could do with another test (looks like in tests it's always triggering). We're also not explicitly returning a value if we trigger the until (which _probably_ means nil, which is probably okay, but it might be better to be strict in our ? methods and explicitly return a false.) You can also see more clearly that the false actually got returned.

KludgeKML commented on app/presenters/corporate_information_page_presenter.rb:
```diff
@@ -0,0 +1,157 @@
+  end
+
+  def organisation_logo(organisation = default_organisation)
+    return nil unless organisation && organisation.dig("details", "logo")
```
I think we don't have a test for if this triggers and returns the nil?

KludgeKML commented on app/presenters/corporate_information_page_presenter.rb:
```diff
@@ -0,0 +1,157 @@
+    organisation = nil
+    if organisations.present?
+      organisation = organisations.detect { |org| org["content_id"] == organisation_content_id }
+      raise "No organisation in links that matches the one specified in details: #{organisation_content_id}" unless organisation
```
We don't test this raise. Is it possible to get here at all with our current content?

leenagupte reviewed (commented):
WIP Review. I got as far as: Add ContentsList concern and tests

deborahchua committed:
Add locale keys for corporate_information_page schema pluralization

Ran rake "consolidation:copy_translation[content_item.schema_name.corporate_information_page, formats.corporate_information_page]"

deborahchua committed:
Update notice component in view

This updates the component to use the new Withdrawable methods
introduced in 4b1d2d4.

deborahchua commented on app/presenters/corporate_information_page_presenter.rb:
```diff
@@ -65,4 +103,55 @@ def executive_order_crest?(organisation)
+    expanded_links_from_content_item("corporate_information_pages")
+  end
+
+  def expanded_links_from_content_item(type)
```
I've done some further refactoring and moved as much as possible into the model.
The remaining methods in the presenter help generate a paragraph which includes links to further information pages if they are available, and need `@view_context` to generate the links. I haven't been able to find a nice way of directly displaying them in view instead.

leenagupte commented on spec/models/corporate_information_page_spec.rb:
```diff
@@ -107,4 +112,12 @@
+
+  describe "#contents_items" do
+    it "includes the corporate information heading in the contents list if corporate informations groups are present" do
+      expect(corporate_information_page.content_store_response["details"]["corporate_information_groups"].present?).to be(true)
```
Should this be checking for the presence of `corporate_information_page.corporate_information_groups` instead?

leenagupte reviewed (commented):
We've already chatted about modelling Organisations in person, but as promised I've reviewed the `ContentsList` concern. 
It's very confusing! So I can understand why you've kept most of the code as it was in government-frontend.

Part of me wonders whether it's worth trying to deconstruct the ContentsList a bit in to it's model and presentational parts.

For example, the decision to `show_contents_list?` is quite presentational, not something the model cares about. Could the model return a structured object of headers? The code in the ContentsList is confusing because as I understand it, much of it won't be reached unless `contents_items` is overwritten.

I'm in two minds about whether it's worth the effort to simplify it now, or leave it for a firebreak. I think perhaps if the system tests are thorough, then the refactoring could be left for "later".

deborahchua committed:
Add page title to view

We want to include the organisation as part of the page title.
In government-frontend the page title was included as part of the
application layout.

Audit trail:
- https://github.com/alphagov/government-frontend/blob/baef09ffcce08cc557a7c01e952824ee803d70f9/app/presenters/corporate_information_page_presenter.rb#L10

leenagupte commented on spec/models/corporate_information_page_spec.rb:
```diff
@@ -30,4 +30,43 @@
+
+  describe "#corporate_information" do
+    it "includes group links that are guids" do
+      presented_groups = described_class.new(content_store_response).corporate_information
```
Perhaps set `subject` at the beginning of the describe block so we don't get presenters and models confused in the tests. I had to double check what type of test I was looking at!

e.g.

```ruby
subject(:corporate_information) { described_class.new(content_store_response).corporate_information }
```

then the expectation can use `expect(corporate_information.groups.first` etc

leenagupte commented on spec/requests/corporate_information_page_spec.rb:
```diff
@@ -0,0 +1,24 @@
+RSpec.describe "Corporate Information Page" do
+  before do
+    content_store_has_example_item("/government/organisations/government-digital-service/about", schema: :corporate_information_page)
+    content_store_has_example_item("/government/organisations/government-digital-service/about/our-governance", schema: :corporate_information_page)
```
Why does the "our-governance" page need to be stubbed?

deborahchua replied:
I had to configure both `/about` and `/about/:slug` to render the show template in routes.rb so I though I should test both routes here.

leenagupte commented on spec/system/corporate_information_page_spec.rb:
```diff
@@ -0,0 +1,127 @@
+RSpec.describe "CorporateInformationPage" do
+  context "when visiting a Corporate Information page" do
+    before do
+      @content_store_response = GovukSchemas::Example.find("corporate_information_page", example_name: "corporate_information_page")
```
I'm surprised rubocop has moaned about this, but it normally recommends using a `let` rather than instance variables in tests.

leenagupte reviewed (commented):
This is making good progress. I've added some recommendations in line.

I think that the next thing we need to think about is the Contents List and how it's modelled and presented. It's something that's going to be used by a lot of document types, so it'll be good to get together as a group and dissect the problem.

deborahchua committed:
Add corporate information group methods to CorporateInformationPage presenter

Audit trail:
- https://github.com/alphagov/government-frontend/blob/f25bb54b3ce6fbe2a61e185c85e6137e808b33c4/app/presenters/content_item/corporate_information_groups.rb

deborahchua committed:
Add corporate information groups methods to CorporateInformationPage model

Similar to the organisation branding methods, these methods are in
a separate module in government-frontend but is not used by any
other document types.

Audit trail:
- https://github.com/alphagov/government-frontend/blob/a91838affca0c907ff94beba7c3bd021ddbd1809/app/presenters/content_item/corporate_information_groups.rb
- https://github.com/alphagov/government-frontend/blob/90cf5386b165b3a342689744feece7c83449b153/test/presenters/corporate_information_page_presenter_test.rb#L42-L79

KludgeKML commented on app/views/corporate_information_page/show.html.erb:
```diff
@@ -1,3 +1,5 @@
+<% content_for :title, "#{@presenter.page_title} - #{content_item.default_organisation.title} - GOV.UK" %>
```
Again, in the original code they're testing if a default organisation is present. If it's possible one won't be, we should be testing too. If it's guaranteed to be there, we don't have to test but we should remove guard code in the previous commit.

KludgeKML commented on app/presenters/corporate_information_page_presenter.rb:
```diff
@@ -1,2 +1,49 @@
 class CorporateInformationPagePresenter < ContentItemPresenter
+  def initialize(content_item, view_context)
```
We can probably get rid of the view context by including the link_helper, a la https://github.com/alphagov/frontend/pull/4687/commits/d94fa1399967c20c524e18fbdb8ae1352ff5c095

KludgeKML replied:
(which in turn means we don't need this initializer at all)

deborahchua replied:
I've updated this now in commit [Add corporate information group methods to CorporateInformationPage presenter](https://github.com/alphagov/frontend/pull/4608/commits/a8448d31cbed05af936a21b2c323be72be4bf802)

KludgeKML commented on app/presenters/corporate_information_page_presenter.rb:
```diff
@@ -1,2 +1,49 @@
+    content_item.corporate_information_pages.any?
+  end
+
+  def further_information_links
+    links = further_information_types.map do |further_information_type|
+      page = content_item.corporate_information_pages.find { |content_item| content_item.document_type == further_information_type }
+      next unless page
+
+      link = @view_context.govuk_styled_link(page.title, path: page.base_path)
+
+      I18n.t("formats.corporate_information_page.#{further_information_type}_html", link:)
+    end
+
+    links.compact.join(" ").html_safe
+  end
+
+  def further_information_types
+    %w[
+      publication_scheme
+      welsh_language_scheme
+      personal_information_charter
+      social_media_use
+      about_our_services
+    ]
+  end
```
Not a blocking comment, but _maybe_ this isn't entirely presentational and it might make sense for it to be in the model. Maybe we should discuss

KludgeKML commented on app/models/concerns/contents_list.rb:
```diff
@@ -0,0 +1,32 @@
+module ContentsList
+  extend ActiveSupport::Concern
+
+  included do
```
I don't think this actually needs to be in an included block, and we don't need to extend ActiveSupport::Concern - default module behaviour allows object method definitions (which this is)

KludgeKML replied:
Actually, I wonder if this should just be a model - it appears to rely on the class it's included in having a body method/attribute, and that's the only thing that gets passed in. Perhaps one to discuss on Monday morning.

KludgeKML reviewed (changes requested):
Very nice, almost complete I think, but I have added a few not particularly serious notes - a couple of model-related ones that it might be best to talk about briefly.

deborahchua commented on app/models/corporate_information_page.rb:
```diff
@@ -1,2 +1,9 @@
-end
\ No newline at end of file
+  def default_organisation
+    return if organisations.blank?
```
I've just checked the schema rules and the [default organisation ID is required](https://github.com/alphagov/publishing-api/blob/main/content_schemas/formats/corporate_information_page.jsonnet#L9) so I don't think we need a check in the view.

I don't think it's possible for organisations to be blank (there should be at least one) as you need an existing organisation to be able to create a corporate information page in Whitehall. I've removed the guard as you've suggested 👍

deborahchua commented on app/models/logo.rb:
```diff
@@ -4,7 +4,7 @@ class Logo
     @image = get_image(logo["image"])
     @crest = logo["crest"]
-    @formatted_title = logo["formatted_title"]
+    @formatted_title = logo["formatted_title"].html_safe
```
Thanks! I've updated in commit [Update title in Logo model](https://github.com/alphagov/frontend/pull/4608/commits/efb5aaac3572a9acbe8b64f9c542ca9a75a0387e)

deborahchua commented on app/controllers/corporate_information_page_controller.rb:
```diff
@@ -1,3 +1,5 @@
 class CorporateInformationPageController < ContentItemsController
-  def show; end
+  def show
+    @presenter = CorporateInformationPagePresenter.new(@content_item, view_context)
```
I've included the LinkHelper in the presenter so we no longer need the `view_context` - I've updated this in commit
[Add basic model, controller and presenter](https://github.com/alphagov/frontend/pull/4608/commits/598f04014359faa54f88ef0ea56c9710a585546e)

KludgeKML commented on spec/models/corporate_information_page_spec.rb:
```diff
@@ -17,4 +17,64 @@
+
+        corporate_information = described_class.new(content_store_response).corporate_information
+
+        expect(corporate_information.first[:links].count).not_to eq(groups.count)
```
Non-blocking note: You could make this `.to eq(groups.count - 1)`, I think? And it would be a bit of a tighter test, because it wouldn't pass if all the links were gone, or there were a thousand additional links? (I'm assuming there's more than one link, of course, so that setting the bogus GUID doesn't take this from 1 link to 0)

KludgeKML commented on app/controllers/corporate_information_page_controller.rb:
```diff
@@ -1,5 +1,6 @@
 class CorporateInformationPageController < ContentItemsController
   def show
     @presenter = CorporateInformationPagePresenter.new(@content_item)
+    @contents_outline_presenter = ContentsOutlinePresenter.new(@presenter.contents_list_headings)
```
I think the preferred way of getting at a ContentsOutlinePresenter is to include it in the main presenter behind a method (as here: https://github.com/alphagov/frontend/blob/main/app/presenters/specialist_document_presenter.rb#L5-L9 ) - I appreciate I originally did have separate presenters when I first altered specialist documents, so this is on me.

deborahchua replied:
No worries @KludgeKML , that makes sense to include in the main presenter! I've updated in https://github.com/alphagov/frontend/pull/4608/commits/c246acece6b021b63a0f5511ca5fa0ddccb989a3 - `contents_list_headings` is now a private method which is called from a new method`headers_for_contents_list_component` similar to that in specialist documents.

KludgeKML commented on spec/requests/corporate_information_page_spec.rb:
```diff
@@ -0,0 +1,24 @@
+RSpec.describe "Corporate Information Page" do
```
We might need an audit trail message for the system/request tests?

deborahchua replied:
Good spot thanks @KludgeKML , I've updated in https://github.com/alphagov/frontend/pull/4608/commits/c21d7fa549aa9856b2c15ebe89dfd7c481c9587a

KludgeKML reviewed (changes requested):
Looking pretty good - I have a few comments, but most of them are about commit messages and one is just something I fear I led you astray on myself.

deborahchua commented on app/controllers/corporate_information_page_controller.rb:
```diff
@@ -0,0 +1,3 @@
+class CorporateInformationPageController < ContentItemsController
```
Ah thank you! Yeah I wish it was possible to comment on just a commit, my workaround is to quote the commit title.

deborahchua replied:
That's now updated in https://github.com/alphagov/frontend/pull/4608/commits/4fc126182f28b607f64352062dd304ae0ccdee36

KludgeKML reviewed (approved):
LGTM!

leenagupte commented on app/models/corporate_information_page.rb:
```diff
@@ -0,0 +1,61 @@
+  def initialize(content_store_response)
+    super(content_store_response)
+
+    @headers = content_store_response.dig("details", "headers")
```
Hmmm, so `headers` aren't being modelled with ContentsOutline?

leenagupte commented on app/presenters/corporate_information_page_presenter.rb:
```diff
@@ -0,0 +1,69 @@
+
+    exclude_nested_headings
+
+    ContentsOutline.new(content_item.headers)
```
Oh, it's modelled here. Perhaps the modelling should be moved to the model for [consistency](https://github.com/alphagov/frontend/blob/main/app/models/specialist_document.rb#L15)?

deborahchua replied:
Ah yes, I did consider this initially, but both the model and presenter uses the method [corporate_information_heading](https://github.com/alphagov/frontend/pull/4608/commits/d45d542fd523fe4af3e71c878232454d9c75c88b#diff-59cebc1a6c28f24f1ed9f80839a0e0a226b94925414b84bfc32fd143d200f356R10) to show the header in the view, and to add it to the `content_item.headers` array.

leenagupte replied:
Do you mean the view and the presenter? rather than model and presenter?

It's not a blocker. Just a question.

deborahchua replied:
I've moved this to the model now in commit [Model headers for contents list using ContentsOutline model,](https://github.com/alphagov/frontend/pull/4608/commits/db22e38f0cb0c2689901b6016eb91d526cc6a561) as it does make more sense 👍

leenagupte commented on app/presenters/corporate_information_page_presenter.rb:
```diff
@@ -0,0 +1,69 @@
+    ContentsOutline.new(content_item.headers)
+  end
+
+  def exclude_nested_headings
```
Is there any restriction on the level? For example, if there are no level 2 headings, and level 3 headings are the top level in the headers hash should the contents list still be displayed?

deborahchua replied:
That is a great point, I hadn't thought of that!

I think I'll have to use [this additional check](https://github.com/alphagov/frontend/blob/4af199c3df5f52b86e37b50664c12e847046c0f1/app/models/contents_outline.rb#L10), similar to that in specialist documents - https://github.com/alphagov/frontend/blob/4af199c3df5f52b86e37b50664c12e847046c0f1/app/presenters/specialist_document_presenter.rb#L30

deborahchua replied:
I've updated this in commit [Update presenter with contents list headers](https://github.com/alphagov/frontend/pull/4608/commits/e830fee696735aad587a87ca1f08394eaec68a85)

leenagupte reviewed (commented):
Following the conversation in Slack I've added a couple of inline thoughts / questions.

deborahchua committed:
Add headers to CorporateInformationPage model

We can then use the headers in the ContentsOutline model to convert
them into a format that can be passed to the contents list component.

This is a new approach to generating contents lists in Frontend,
and significantly reduces the complexity that exists in government-frontend:
- app/models/concerns/contents_list.rb > https://github.com/alphagov/government-frontend/blob/90cf5386b165b3a342689744feece7c83449b153/app/presenters/content_item/contents_list.rb
- spec/support/concerns/contents_list.rb > https://github.com/alphagov/government-frontend/blob/90cf5386b165b3a342689744feece7c83449b153/test/presenters/content_item/contents_list_test.rb

References:
- alphagov/whitehall#10143
- alphagov/publishing-api#3280

deborahchua committed:
Update presenter with contents list headers

This is a new approach to generating contents lists in Frontend,
and significantly reduces the complexity that exists in government-frontend.
We have not included the existing rules and restrictions; the contents
list will always be shown if there are H2 headings present.

Corporate information pages also have an additional H2 heading,
"Corporate information", if there are `corporate_information_groups`
available in the `details` hash. This is a section at the bottom of
the page which include  additional links to other corporate information
pages. This heading is added to the end of the `content_item.headers` array.

In addition, we also don't include nested headings (H3 and below).

Once we have our array of H2 headings, we convert the headers using
the ContentsOutline model and presenter into a format suitable to
then pass to the contents_list_with_body component.

The methods `corporate_information_heading` and `corporate_information_heading_id`
have been combined and updated so that they are in a format suitable
for the ContentsOutline model, and available to use in the view.

Audit trail:
- https://github.com/alphagov/government-frontend/blob/4c38f1d3d8600373fe49ab506c758ee05e4909a3/app/presenters/corporate_information_page_presenter.rb#L21-L31
- https://github.com/alphagov/government-frontend/blob/4c38f1d3d8600373fe49ab506c758ee05e4909a3/app/presenters/content_item/corporate_information_groups.rb#L50-L57
- https://github.com/alphagov/government-frontend/blob/4c38f1d3d8600373fe49ab506c758ee05e4909a3/app/presenters/content_item/contents_list.rb#L11
- https://github.com/alphagov/government-frontend/blob/4c38f1d3d8600373fe49ab506c758ee05e4909a3/test/presenters/corporate_information_page_presenter_test.rb#L42-L49
- https://github.com/alphagov/government-frontend/blob/90cf5386b165b3a342689744feece7c83449b153/test/presenters/content_item/contents_list_test.rb

KludgeKML reviewed (approved):
LGTM

AshGDS reviewed (dismissed):
EDIT: Ignore this, apologies, was looking at an older commit.

AshGDS reviewed (approved):
LGTM 👍
